azure-pipelines.yml

```diff
@@ -0,0 +1,49 @@
+# This pipeline is NEVER triggered automatically.
+# This pipeline is used for initialization of a new (or cleared) kubernetes cluster.
+# Running this pipeline unnecessarily may cause the ingress IP address to reset.
+# The ingress IP address is how you access the application, so...
+# BE CAREFUL!!!
+trigger: none
+
+pool:
+  vmImage: ubuntu-latest
+
+variables:
+  # to initialize the staging environment k8sServiceConnection should be 'k8s-team1-config-staging'
+  # to initialize the development environment k8sServiceConnection should be 'k8s-team1-config-dev'
+  k8sServiceConnection: 'k8s-team1-config-staging'
```

Comment:
1. then make this a parameter, not a variable. pipeline parameters can be set in the GUI at run-time, and you can set `values` in the parameter definitions so that there will be a dropdown to pick between the two allowed values.
2. with a deployment job and environments, we could record these deployments better.

Reply:
To set up an Environment we have to have a login for the associated Azure Subscription.  When you go to create an environment you get a drop-down with subscriptions available (only my personal subscription is there) and when you select it an Azure login window pops up requesting your username and password.

Reply:
I didn't know that you could use parameters outside of a template.  I have changed it per your suggestion.

Reply:
>To set up an Environment we have to have a login for the associated Azure Subscription

this is only the default way to add a kubernetes resource. not all clusters are on aks, after all, and azure wouldn't be that restrictive. there's a dropdown to provide the cluster details more manually.

Reply:
I tried setting up a generic provider, but when I followed the instructions I get a Server URL that starts with `http://127.0.0.1`, so the validation fails, since it is on localhost.

Reply:
the instructions rely on you having the cluster in question as your current-context when running those commands

Reply:
I was running them through the terminal in lens with the dev cluster active.

Reply:
odd. i can troubleshoot that with you if you want to spend the time on it.

Reply:
also, you can set up empty (no resources) environments that serve only to record the history of deployment from builds, and those still serve a purpose.
it's dumb you can't create a cluster resource based on an existing service connection.